### PR TITLE
Migrate documentation for `regr*` aggregate functions to code

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/12827

## Rationale for this change

https://github.com/apache/datafusion/pull/12861 from @jonathanc-n moves some of the aggregate function documentation to the code, but the `regr` functions are handled specially

## What changes are included in this PR?

Move `regr_*` aggregate functions to code

## Are these changes tested?
Yes y CI

### CODE REVIEW

File: datafusion/functions-aggregate/src/regr.rs [hunk 1]
```diff
@@ -135,6 +123,148 @@ pub enum RegrType {
...
+        hash_map.insert(
+            RegrType::Slope,
+            Documentation::builder()
+                .with_doc_section(DOC_SECTION_STATISTICAL)
+                .with_description(
+                    "Returns the slope of the linear regression line for non-null pairs in aggregate columns. \
+                    Given input column Y and X: regr_slope(Y, X) returns the slope (k in Y = k*X + b) using minimal RSS fitting.",
+                )
+                .with_syntax_example("regr_slope(expression_y, expression_x)")
+                .with_standard_argument("expression_y", "Expression")
```
Comment (Omega359) [inline datafusion/functions-aggregate/src/regr.rs:146]: This currently renders as ```Expression expression to operate on. Can be a constant, column, or function, and any combination of operators.``

**Comment by Omega359** [comment]
Beyond the one minor change this LGTM

File: datafusion/functions-aggregate/src/regr.rs (hunk 1 above)
Comment (alamb) [reply datafusion/functions-aggregate/src/regr.rs]: Fixed. Thank you

**Review by korowa — APPROVED** [review]
LGTM, thank you @alamb

**Comment by alamb** [comment]
Thank you for the review @korowa

**Review by goldmedal — APPROVED** [review]
Thanks @alamb  it looks good to me 👍

**Comment by alamb** [comment]
🚀 let's do this. Thanks again @goldmedal

**Pull request merged**